internal/config/config.go

```diff
@@ -45,6 +46,7 @@ const reportingDelayEnv = "REPORTING_RATE"
 const reportingTimeoutEnv = "REPORTING_TIMEOUT"
 const verboseEnv = "VERBOSE"
 const httpTracingEnv = "HTTP_TRACING"
+const failFastEnv = "SPLUNK_EXTENSION_FAIL_FAST"
```

Comment:
The env var is not documented in https://github.com/signalfx/gdi-specification/blob/main/specification/configuration.md

> If a new configuration variable is needed by a GDI repository it MUST be brought to the GDI specification as a GitHub issue.

Reply:
Can you add a hyperlink to the GH issue so that we make sure it is not lost and we will have a reference?

Also, I think the env var should be documented as "experimental"

Reply:
> If a new configuration variable is needed by a GDI repository it MUST be brought to the GDI specification as a GitHub issue.

@flands @MrAlias How "strict" are we on this requirement? Do we only need a GitHub issue? Do we need to have it defined in the specification? If so then on the `main` branch is enough or does it have to be released?

Reply:
> How "strict" are we on this requirement? 

A GitHub issue is needed to track this addition on a provisional basis. I can be added once that is done, with the following caveats:

> If a GDI repository requires an immediate configuration variable that is not available in the OpenTelemetry specification and not available in the GDI specification, the repository MAY introduce a repository-specific configuration variable until the GDI specification maintainers make a decision. Any repository-specific configuration variable defined SHOULD be prefixed with SPLUNK_ and MUST NOT be marked as stable unless added to the GDI specification. Upon resolution by the GDI specification, the GDI repository MUST change the repository-specific configuration variable by the repository's next minor release. This change MAY result in a breaking change so caution should be exhibited when considering repository-specific configuration variables.

Reply:
That said, I am hesitant to accept such a change. The GDI specification already has [normative requirements](https://github.com/signalfx/gdi-specification/blob/main/specification/configuration.md#serverless) on performance when a configuration value is missing:

> As there is no deployment phase in case of Serverless functions, if a required configuration property is missing, the serverless instrumentation library MUST log an error but MUST still execute.

I would guess that this behavior is expect for all operations and the current behavior of this library (failing the workflow) is seen as a bug. In that case I would be hesitant in adding this environment variable as it will likely not be accepted at the GDI level and might need to be removed here.

Do we have an understanding of the behavior for lambda extensions of other languages?

Reply:
@MrAlias there is only one extension - this one. External extensions are language-agnostic. 
The extension as it operates now (fail-fast) has been in the wild for a long time (long before we - instrumentals - took over). There may be customers relying on that (eg for automation / alerting). Therefore I believe that:
- changing the behaviour via a config property
- keeping it internal for the extension
is the right way to move forward.

Reply:
IMO addressed here https://github.com/signalfx/splunk-extension-wrapper/pull/92/commits/1cad916c9b24befd06b81ce54af7b3d150d6bbb6